engine/src/main/java/org/hibernate/search/util/impl/ReflectionHelper.java

```diff
@@ -200,6 +200,36 @@ public static Object createInstance(Class<?> clazz, boolean checkForFactoryAnnot
 		return instance;
 	}
 
+	/**
+	 * Checks whether the specified type is a floating point type.
+	 *
+	 * @param type the type to check
+	 * @return {@code true} if the specified type is an floating point type or a wrapper thereof. {@code false} otherwise.
+	 */
+	public static boolean isFloatingPointType(Class<?> type) {
+		return float.class.equals( type )
```

Comment:
So btw. one cannot do facet queries e.g. on `BigDecimal`? Is this a limitation in Lucene or on our side? It seems interesting when it comes to prices for instance.